### PR TITLE
Add analyzer for duplicate data row

### LINKED ISSUE #5085: Analyzer suggestion: Duplicate DataRow
An analyzer should detect if two `DataRow` attributes on the same test method are providing the same data. This should be a typo.

AB#2404444

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #5085

### CODE REVIEW

File: src/Analyzers/MSTest.Analyzers/DuplicateDataRowAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        foreach (AttributeData attribute in methodSymbol.GetAttributes())
+        {
+            if (!dataRowAttribute.Equals(attribute.AttributeClass, SymbolEqualityComparer.Default))
+            {
+                continue;
+            }
+
+            if (!dataRowArguments.Add(attribute.ConstructorArguments))
+            {
+                context.ReportDiagnostic((attribute.ApplicationSyntaxReference?.GetSyntax(context.CancellationToken).GetLocation() ?? Location.None).CreateDiagnostic(Rule));
```
Comment (Evangelink) [inline src/Analyzers/MSTest.Analyzers/DuplicateDataRowAnalyzer.cs]: Is there no helper for that on roslyn-analyzers? Or maybe it's already available in the code I copied.

File: src/Analyzers/MSTest.Analyzers/Resources.resx [hunk 2]
```diff
@@ -573,4 +573,10 @@ The type declaring these methods should also respect the following rules:
   <data name="UseConditionBaseWithTestClassMessageFormat" xml:space="preserve">
     <value>The attribute '{0}' which derives from 'ConditionBaseAttribute' should be used only on classes marked with `TestClassAttribute`</value>
   </data>
+  <data name="DuplicateDataRowTitle" xml:space="preserve">
+    <value>A duplicate 'DataRow' attribute is found</value>
```
Comment (Evangelink) [inline src/Analyzers/MSTest.Analyzers/Resources.resx]: Maybe something like `Avoid duplicated 'DataRow' entries`? or `Do not duplicate 'DataRow' entries`

File: src/Analyzers/MSTest.Analyzers/Resources.resx [hunk 3]
```diff
@@ -573,4 +573,10 @@ The type declaring these methods should also respect the following rules:
...
+  </data>
+  <data name="DuplicateDataRowMessageFormat" xml:space="preserve">
+    <value>Do not duplicate 'DataRow' attributes. This is usually a copy/paste error.</value>
```
Comment (Evangelink) [inline src/Analyzers/MSTest.Analyzers/Resources.resx]: Just wondering if we should point out the indexes of duplicated entries

**Review by Evangelink — COMMENTED** [review]
Please add a test where you have 2 pairs of duplicate attributes on the same method
